.github/workflows/test-on-iotlab.yml

```diff
@@ -0,0 +1,129 @@
+  # Allow manual trigger
+  workflow_dispatch:
+    inputs:
+      riot_version:
+        description: 'RIOT version to checkout'
+        required: true
+        default: 'master'
+
+jobs:
+  # Runs all tests on IoT-LAB boards
+  compile_and_test_for_board:
+    strategy:
+      max-parallel: 8
+      fail-fast: false
+      matrix:
+        boards:
+          - riot: iotlab-m3
+            iotlab:
+              archi: m3:at86rf231
+              site: saclay
+          - riot: nrf51dk
+            iotlab:
+              archi: nrf51dk:ble
+              site: saclay
+          - riot: nrf52dk
+            iotlab:
+              archi: nrf52dk:ble
+              site: saclay
+          - riot: nrf52840dk
+            iotlab:
+              archi: nrf52840dk:multi
+              site: saclay
+          - riot: nrf52832-mdk
+            iotlab:
+              archi: nrf52832mdk:ble
+              site: saclay
+          - riot: samr21-xpro
+            iotlab:
+              archi: samr21:at86rf233
+              site: saclay
+          - riot: b-l072z-lrwan1
+            iotlab:
+              archi: st-lrwan1:sx1276
+              site: saclay
+          - riot: b-l475e-iot01a
+            iotlab:
+              archi: st-iotnode:multi
+              site: saclay
+    runs-on: ubuntu-latest
+    name: ${{ matrix.boards.riot }}
+    env:
+      IOTLAB_NODE: auto-ssh
+      BUILD_IN_DOCKER: 1
+      # Force .bin files generation because these files are used to flash on IoT-LAB and
+      # because compile_and_test_for_board forces RIOT_CI_BUILD which skip .bin
+      # files generation
+      DOCKER_ENVIRONMENT_CMDLINE: -e BUILD_FILES=\$$\(BINFILE\)
+      COMPILE_AND_TEST_FOR_BOARD: ./dist/tools/compile_and_test_for_board/compile_and_test_for_board.py
+      COMPILE_AND_TEST_ARGS: --with-test-only --jobs=2
+    steps:
+      - name: Set up Python 3.8
+        uses: actions/setup-python@v2
+        with:
+          python-version: 3.8
+      - name: Install Python dependencies
+        run: |
+          python -m pip install --upgrade pip
+          pip install iotlabcli pexpect
+      - name: Configure credentials
+        run: echo '${{ secrets.IOTLABRC }}' > ~/.iotlabrc
+      - name: Setup SSH agent
+        uses: webfactory/ssh-agent@v0.4.0
+        with:
+          ssh-private-key: ${{ secrets.IOTLAB_PRIVATE_KEY }}
+      - name: Fetch host key from IoT-LAB ${{ matrix.boards.iotlab.site }} site
+        run: |
+          IOTLAB_USER=$(cat ~/.iotlabrc | cut -f1 -d:)
+          ssh -oStrictHostKeyChecking=accept-new \
+          ${IOTLAB_USER}@${{ matrix.boards.iotlab.site }}.iot-lab.info exit
+      - name: Pull riotbuild docker image
+        run: |
+          docker pull riot/riotbuild:latest
+      - name: Checkout RIOT
+        uses: actions/checkout@v2
+        with:
+          ref: ${{ github.event.inputs.riot_version }}
+      - name: Print RIOT version
+        run: |
+          git log -1 --pretty
```

Comment:
The [checkout action does this already for you](https://github.com/aabadie/RIOT/runs/926488305?check_suite_focus=true#step:8:457) though not as pretty ;-). Still, I think this information is redundant.

Reply:
Left-over from my previous tests...